src/m365/pp/commands/solution/solution-get.ts

```diff
@@ -0,0 +1,125 @@
+import { AxiosRequestConfig } from 'axios';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import PowerPlatformCommand from '../../../base/PowerPlatformCommand';
+import commands from '../../commands';
+import { Publisher, Solution } from './Solution';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string;
+  id?: string;
+  name?: string;
+  asAdmin: boolean;
+}
+
+class PpSolutionGetCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return commands.SOLUTION_GET;
+  }
+
+  public get description(): string {
+    return 'Lists a specific solution in a given environment.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['uniquename', 'version', 'publisher'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environment <environment>'
+      },
+      {
+        option: '-i, --id'
+      },
+      {
+        option: '-n, --name'
```

Comment:
Name option is not a flag.
The difference between this and `asAdmin` is that we expect the user to enter a value for option `name` while for `asAdmin` we don't expect a value at all `pp solution get --environment abc --name def --asAdmin`.

```suggestion
        option: '-n, --name [name]'
```